### DISCUSSION TITLE
pre-receive webhook on github enterprises

### DISCUSSION THREAD

**Discussion by rakesh0022** [community/community#86622] [Enterprise]
### Select Topic Area

Question

### Body

Hi Team,

I have setup a pre-receive hooks on my github enterprises and created a hooks for various repos ,   when i disable the hooks and then enable it, then it will work for 2-3 minutes and then again it will stop working, then again if i disable and enable the hooks it will work for 2-3 minutes and then stop working. 
Could you please help what seems to be issue here ?

**Comment by Naimulnobel** [comment]
The issue you're experiencing could be due to a variety of factors. Here are a few possibilities:

1. **Rate Limiting**: GitHub Enterprise has rate limits for webhook events. If your hook is being triggered too frequently, it might hit this limit and stop working.

2. **Timeouts**: If your hook takes too long to execute, GitHub might stop sending requests to it. Ensure that your hook responds quickly to avoid this.

3. **Server Errors**: If your hook is returning HTTP error statuses (like 500), GitHub might disable it. Check your server logs to see if there are any unhandled errors.

4. **Network Issues**: There might be network issues between GitHub and the server where your hook is hosted. Check for any network outages or disruptions.

5. **GitHub Enterprise Settings**: There might be settings in your GitHub Enterprise instance that are causing the hooks to be disabled. Check with your GitHub Enterprise administrator.

To debug this issue, you can check the delivery logs for your hook in the GitHub UI. This will show you the HTTP response code for each delivery, which can help you identify what's going wrong. If you're still unable to resolve the issue, you might want to contact GitHub Enterprise Support for further assistance.
